File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        if head==None:
            return False
        slow=head
        fast=head
        while fast.next!=None and fast.next.next!=None:
            fast=fast.next.next
            slow=slow.next
           
            if slow== fast:
                return True
           
        return False

        

File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    Set<ListNode> s=new HashSet<>();
    boolean res=false;
    public void dFS(ListNode node){
        if(node==null)
            return;
        if(s.contains(node)){
            res=true;
            return;
        }
        s.add(node);
        dFS(node.next);


    }
        

    public boolean hasCycle(ListNode head) {
        dFS(head);
        return res;

        
    }
}